File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/common/valueobjects/ChannelType.java
package com.enterpriseagentos.backendjava.domain.common.valueobjects;

import java.util.Locale;

public enum ChannelType {
    Slack,
    Telegram,
    Discord,
    WhatsApp,
    Teams,
    GoogleChat;

    public String toWire() {
        return switch (this) {
            case Slack -> "slack";
            case Telegram -> "telegram";
            case Discord -> "discord";
            case WhatsApp -> "whatsapp";
            case Teams -> "teams";
            case GoogleChat -> "google-chat";
        };
    }

    public static ChannelType fromWire(String value) {
        return switch (value.trim().toLowerCase(Locale.ROOT)) {
            case "slack" -> Slack;
            case "telegram" -> Telegram;
            case "discord" -> Discord;
            case "whatsapp" -> WhatsApp;
            case "teams" -> Teams;
            case "google-chat" -> GoogleChat;
            default -> throw new IllegalArgumentException("Unknown channel type: " + value);
        };
    }
}

File: apps/backend/src/Common/Domain/ValueObjects/ChannelType.cs
namespace EnterpriseAgentOs.Domain.Common.ValueObjects;

public enum ChannelType
{
    Slack,
    Telegram,
    Discord,
    WhatsApp,
    Teams,
    GoogleChat,
}

public static class ChannelTypeExtensions
{
    public static string ToStorageString(this ChannelType type) => type switch
    {
        ChannelType.Slack => "slack",
        ChannelType.Telegram => "telegram",
        ChannelType.Discord => "discord",
        ChannelType.WhatsApp => "whatsapp",
        ChannelType.Teams => "teams",
        ChannelType.GoogleChat => "google-chat",
        _ => throw new ArgumentOutOfRangeException(nameof(type)),
    };

    public static ChannelType ToChannelType(this string value) => value.ToLowerInvariant() switch
    {
        "slack" => ChannelType.Slack,
        "telegram" => ChannelType.Telegram,
        "discord" => ChannelType.Discord,
        "whatsapp" => ChannelType.WhatsApp,
        "teams" => ChannelType.Teams,
        "google-chat" => ChannelType.GoogleChat,
        _ => throw new ArgumentOutOfRangeException(nameof(value), $"Unknown channel type: {value}"),
    };
}
